bids/modeling/statsmodels.py

```diff
@@ -337,21 +337,24 @@ def _build_groups(objects, group_by):
 
         groups = defaultdict(list)
 
-        # sanitize grouping entities, otherwise weird things can happen
-        group_by = list(set(group_by) & VALID_GROUPING_ENTITIES)
-
         # Get unique values in each grouping variable and construct indexing DF
         entities = [obj.entities for obj in objects]
         df = pd.DataFrame.from_records(entities)
 
+        # Single-run tasks and single-session subjects may not have entities
+        dummy_groups = {"run", "session"} - set(df.columns)
+
```

Comment:
Check for all possible valid grouping entities and return an error is user tries to group on something not allowed.